### ISSUE TITLE
A problem with external storage permissions

### ISSUE THREAD

**Issue by DCNick3** [cryptomilk/android_device_sony_lilac#46]
I've got a problem with external storage permissions. In partucular - with access to path returned by `Context.getExternalFilesDir(null)`. As per docs, application is not required to have `WRITE_EXTERNAL_STORAGE` or `READ_EXTERNAL_STORAGE` permissions to access that, but an app fails with "Access Denied" while doing that:

```
10-20 21:57:35.932 28109 28109 E AndroidRuntime: FATAL EXCEPTION: main
10-20 21:57:35.932 28109 28109 E AndroidRuntime: Process: com.pavelrekun.rekado, PID: 28109
10-20 21:57:35.932 28109 28109 E AndroidRuntime: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:503)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
10-20 21:57:35.932 28109 28109 E AndroidRuntime: Caused by: java.lang.reflect.InvocationTargetException
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        ... 1 more
10-20 21:57:35.932 28109 28109 E AndroidRuntime: Caused by: java.io.FileNotFoundException: /storage/emulated/0/Android/data/com.pavelrekun.rekado/files/sx_loader.bin (Permission denied)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at java.io.FileOutputStream.open0(Native Method)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at java.io.FileOutputStream.open(FileOutputStream.java:308)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at java.io.FileOutputStream.<init>(FileOutputStream.java:238)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at java.io.FileOutputStream.<init>(FileOutputStream.java:180)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.pavelrekun.rekado.services.extensions.GenericExtensionsKt.toFile(GenericExtensions.kt:15)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.pavelrekun.rekado.services.utils.MemoryUtils.copyAsset(MemoryUtils.kt:21)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.pavelrekun.rekado.services.utils.MemoryUtils.copyBundledPayloads(MemoryUtils.kt:12)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.pavelrekun.rekado.screens.payload_fragment.PayloadsView.initList(PayloadsView.kt:41)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.pavelrekun.rekado.screens.payload_fragment.PayloadsView.initViews(PayloadsView.kt:34)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at com.pavelrekun.rekado.screens.payload_fragment.PayloadsFragment.onViewCreated(PayloadsFragment.kt:31)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:892)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:439)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.FragmentManagerImpl.executeOps(FragmentManagerImpl.java:2076)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1866)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1821)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1727)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at androidx.fragment.app.FragmentManagerImpl$2.run(FragmentManagerImpl.java:150)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:873)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:193)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6718)
10-20 21:57:35.932 28109 28109 E AndroidRuntime:        ... 3 more
```

Where `/storage/emulated/0/Android/data/com.pavelrekun.rekado/files/` is external storage path for `com.pavelrekun.rekado`.

Please, look into that.

**Comment by cryptomilk** [comment]
You should report that upstream to LineageOS.

**Comment by DCNick3** [comment]
Not sure if that is not specific to this particular build. At xda there is information about workaround for some apps:

> Some apps don't work (Gardenscapes, Die Zeit, ...)
>
>When an app is installed then it should ask for Permissions on the first startup. However for some apps this doesn't happen. To fix it you need to go to Settings -> Apps & Notifications -> See all apps -> Select the App -> Permissions. Now manually activate the required permission, normally the Storage Permission is missing in that case! 

As far as I can see, this may have to do with my issue, as it is fixed by giving storage permission to an app. Does this problem exist on upstream?

**Comment by YamiYukiSenpai** [comment]
I'm now unable to download anything to my SD card, especially Nextcloud, Kindle, etc.

**Comment by DCNick3** [comment]
I think your issue does not have to do anything with mine. Separate issue should be opened.

**Comment by cryptomilk** [comment]
This is really strange. Hmm.

**Comment by derfelot** [comment]
@cryptomilk this should also be fixed with new build. The LOS recorder on 17.1 had the same issue with Context.getExternalFilesDir() and is also working now.

**Comment by cryptomilk** [comment]
Should be fixed with v1.2
